### DISCUSSION TITLE
Is there a default time out for docker or python when self hosting Timetagger?

### DISCUSSION THREAD

**Discussion by Tagirijus** [almarklein/timetagger#479] [Q&A]
Hey there,

two times now already I experienced that the running docker container seem to have stopped suddenly, thus being Timetagger not available anymore for the client. The time span it ran was different. Once it stopped after <12 h _(while it could also be that it was me by accident ... I know ... not the best info base :D )_ - AFAIk the docker log for the project was empty when I started it manually again. The other time the docker ran for around 40h. I had to start the docker container again manually - yet this time the docker log sad "Server shutting down" as if I would have stopped it manually (I guess).

Since I am still learning some server related things I wanted to ask here:

1. Is this a normal behavior that the docker container stops after some time? Or the python script or so?
2. If so: what would be the best practice to let Timetagger run longer? Or maybe a cron job to restart it from time to time automatically?

_Here are some rough specs of the server: it's running Ubuntu, according to the provider it gives me 2 CPU cores and 4 GB RAM. According to the resource graph running Timetagger almost needs nothing for those resources. It is a cloud server by Hetzner, which is said (according to people who recommended this provider) to have excellent virtualization._

I sense as if I might be doing something wrong here. Worst case would be that my rent server might have some trouble with it? Thanks for any help in advance.

Also sorry, if I might have missed giving some important information. I am also still doing some research in parallel, yet I also wanted to post the question here. Thanks! (=

**Comment by Tagirijus** [comment]
Thanks for your reply. I did not change any default configuration of anything so far. Also for now the docker container is running for 3 days straight again. I will try to investigate further, if I am able to set up some kind of logging, which might be able to give me hints on why the docker container might have stopped _(if it will happen again)_. When running `docker logs [timetagger_container]` I only got the one _"Server shutting down"_ line last time.

Further replies are welcome, of course. Have a nice day, everyone! (=

**Comment by almarklein** [comment]
Timetagger does not shut itself down, so I'm not sure why this is happening. The "Server shutting down" message indicates that the python process, shut down cleanly, most likely because the docker container was stopped. But I don't know why that would happen. I haven't seen this myself.

**Comment by Tagirijus** [comment]
For now the container is running for 4 days straight. If I look back ... maybe it had something to do with updating anything. Maybe I've missed that some updates required docker to be restarted or so. I will keep an eye on that. Thanks for the reply. Good to know that _"Server shutting down"_ means that it shut down cleanly!
